From: "Barak A. Pearlmutter" <bap@debian.org>
Date: Mon, 24 Jun 2013 18:05:04 +0100
Subject: maxint

Via: git://pkgs.fedoraproject.org/mhash.git
Taken from openpkg:
http://www.mail-archive.com/openpkg-cvs@openpkg.org/msg26353.html
---
 lib/stdfns.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/lib/stdfns.c b/lib/stdfns.c
index d2132f6..79b9592 100644
--- a/lib/stdfns.c
+++ b/lib/stdfns.c
@@ -24,6 +24,7 @@
  */
 
 #include "libdefs.h"
+#include <limits.h>
 
 /**
  * Some of these are wrappers. The idea is to eventually produce an extremely
@@ -408,11 +409,11 @@ mutils_memcmp(__const void *s1, const void *s2, const mutils_word32 n)
 		{
 			return(0);
 		}
-		return(-MAXINT);
+		return(-INT_MAX);
 	}
 	if (s2 == NULL)
 	{
-		return(MAXINT);
+		return(INT_MAX);
 	}
 
 	return(memcmp(s1, s2, n));
@@ -539,11 +540,11 @@ mutils_strcmp(__const mutils_word8 *src1, const mutils_word8 *src2)
 		{
 			return(0);
 		}
-		return(-MAXINT);
+		return(-INT_MAX);
 	}
 	if (src2 == NULL)
 	{
-		return(MAXINT);
+		return(INT_MAX);
 	}
 	return(strcmp((char *) src1, (char *) src2));
 }
@@ -562,11 +563,11 @@ mutils_strncmp(__const mutils_word8 *src1, const mutils_word8 *src2, const mutil
 		{
 			return(0);
 		}
-		return(-MAXINT);
+		return(-INT_MAX);
 	}
 	if (src2 == NULL)
 	{
-		return(MAXINT);
+		return(INT_MAX);
 	}
 	return(strncmp((char *) src1, (char *) src2, n));
 }
